### PR TITLE
style(eslint): Add eslint-plugin-ramda

### PR DESCRIPTION
**Pull request by Undistraction**
See: https://github.com/ramda/eslint-plugin-ramda which adds a number of Ramda-related rules.

Note that this will currently fail. I've created a PR so you can take a look and decide which rules we want to enable. I'll happily go through and refactor when this is decided to get this PR passing.

Ref #306

### CODE REVIEW

**Comment by char0n** [comment]
Let's go for it. Please drop the `package-lock.json`, and ammend the codebase to conform to all the rules which will fix all the 10 pending linting issues. We will disable some rules in future on use-case basis only, if we find some rules to be contraproductive.

One last thing. The configure in example looks different that it looks in this PR.

```js
{
  "name": "my-awesome-project",
  "eslintConfig": {
    "plugins": [
      "ramda"
    ],
    "extends": "plugin:ramda/recommended"
  }
}
```

**Comment by Undistraction** [comment]
We are using a separate `.eslintrc` file so it doesn't make sense to copy the config in the project's README. It's better to keep all our eslint config in one place.

File: test/liftFN.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
+import { sum } from 'ramda';
```
Comment (char0n) [inline test/liftFN.js]:
```js
import * as RA from 'ramda';
```

File: test/neither.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { T } from 'ramda';
```
Comment (char0n) [inline test/neither.js]:
```js
import * as RA from 'ramda';
```

File: test/notBoth.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import { F } from 'ramda';
```
Comment (char0n) [inline test/notBoth.js]:
```js
import * as RA from 'ramda';
```

Comment (char0n) [reply test/notBoth.js]: and  use `R.F` to that we don't fonfuse RA and R functions inside tests.

**Comment by char0n** [comment]
Please rebase on top of master and resolve merge conflicts.

**Comment by Undistraction** [comment]
I added a few more rules which are disabled in the default recommended configuration. They all seem sensible and didn't require flag any errors.

**Comment by Undistraction** [comment]
Looks like Code Climate whitelists eslint plugins and this plugin hasn't been whitelisted (yet).

**Comment by char0n** [comment]
That is unfortunate. Can we do something about that ? Adding exception for every line of code isn't a way to go ;]

**Comment by Undistraction** [comment]
I'm looking into it at the moment.

**Comment by guillaumearm** [comment]
Ref: codeclimate/codeclimate-eslint#369

**Comment by Undistraction** [comment]
Unfortunately I think it's a matter of sitting on this until Code Climate add the plugin to their supported list.

**Comment by char0n** [comment]
Ok we will wait. We have an issue where we track it ? Close this PR for
time beeing and thx for the research

**Comment by Undistraction** [comment]
Closing because we are blocked.

**Pull request closed without merging**